app/controllers/api/v1/facts_controller.rb

```diff
@@ -0,0 +1,87 @@
+class Api::V1::FactsController < ApplicationController
+  include AuthenticationCheck
+
+  before_action :is_user_logged_in
+  before_action :check_access, only: [:show, :update, :destroy]
+  before_action :set_member, only: [:index, :create]
+  before_action :set_fact, only: [:show, :update, :destroy]
+  # skip_before_action :verify_authenticity_token
+  
+
+
+  # GET /members/:member_id/facts
+  def index
+    render json: { facts: @member.facts } # note that because the facts route is nested inside members
+    # we return only the facts belonging to that member
+  end
+
+  # GET /members/:member_id/facts/:id
+  def show
+    # your code goes here
+      # your code goes here
+    if @fact
+      render json: { fact: @fact }
+    else
+      render json: { error: "Fact Not Found" }, status: :not_found
+    end
+  end
+
+
+  # POST /members/:member_id/facts
+  def create
+  
+    @fact = @member.facts.new(fact_params)
+    if @fact.save
+      render json: @fact, status: 201
+    else
+      render json: { error: 
+"The fact entry could not be created. #{@fact.errors.full_messages.to_sentence}"},
+      status: 400
+    end
+  end
+
+  # PUT /members/:member_id/facts/:id
+  def update
+    # your code goes here
+      # your code goes here
+    if @fact.update(fact_params)
+      render json: {message: "Fact record updated successfully."}, status: 200
+    else
+      render json: {error: "Unable to update fact: #{@fact.errors.full_messages.to_sentence}"}, status: 400
+    end   
+  end
+
+  # DELETE /members/:member_id/facts/:id
+  def destroy
+    # your code goes here
+    if @fact
+      if @fact.destroy
+        render json: { message: 'Fact record successfully deleted.'}, status: 200
+      end
+    else
+      render json: { error: 'Fact not found'}, status: 400
+    end
+  end
+
+  private
+
+  def set_member
+    @member = Member.find(params[:member_id])
+  end
+
+  def fact_params
+    params.require(:fact).permit(:fact_text, :likes)
+  end
+
+  def set_fact
+    Rails.logger.debug("Fact ID: #{params[:id]}, Member ID: #{params[:member_id]}")
+    @fact = Fact.find_by(id: params[:id], member_id: params[:member_id])
+  end
+  
+  def check_access 
+    @member = Member.find(params[:member_id])
```

Comment:
There's a strange pattern going on here: some of the actions rely on `@member` being set, and the two `before_action` hooks `set_member` and `check_access` both assign `@member`. My guess is that you carefully chose which `only` values to set on these `before_action`'s such that no action would repeat any `@member` assignments. While this works, I'd say that it's not robust; hypothetically, if others had to make changes to this controller and add more actions, it would be easy to accidentally duplicate `@member` assignments. (This is a fault of the lesson and not yours, but I'd say that `check_access` should not be assigning `@member` because the assignment is not clear in the method name at all.)

One way to efficiently grab a Member object only when you need it, and not have to care about which `before_action`s have run, is to define a private method that checks if we've already fetched a Member object before, return it if so, or otherwise do the fetch. This is called memoization.

```ruby
def member
  @member ||= Member.find(params[:member_id])
end
```

With this method, you can remove the other 
`@member` assignments, and replace all other references of `@member` with `member`. That would make things simpler while staying performant.